packages/commonwealth/server/CommonwealthConsumer/messageProcessors/snapshotConsumer.ts

```diff
@@ -0,0 +1,91 @@
+import axios from 'axios';
+import {factory, formatFilename} from 'common-common/src/logging';
+import {SnapshotNotification} from '../../../shared/types';
+import {DB} from "../../models";
+
+
+export async function processSnapshotMessage(this: { models: DB }, data: SnapshotNotification) {
+  const log = factory.getLogger(formatFilename(__filename));
+  const {space, id, title, body, choices, start, expire} = data;
+
+  const eventType = data.event;
+  let proposal = await this.models.SnapshotProposal.findOne({
+    where: {id: data.id},
+  });
+
+  await this.models.SnapshotSpace.findOrCreate({
+    where: {snapshot_space: space},
+  });
+
+  if (eventType === 'proposal/created' && proposal) {
+    log.error(`Proposal already exists, cannot create`);
+    return;
+  }
+
+  if (!proposal) {
+    proposal = await this.models.SnapshotProposal.create({
+      id,
+      title,
+      body,
+      choices,
+      space,
+      start,
+      expire,
+    });
+  }
+
+  if (eventType === 'proposal/deleted') {
+    console.log('Deleting proposal');
+    await proposal.destroy();
+  }
+
+  const associatedCommunities = await this.models.CommunitySnapshotSpaces.findAll({
+    where: {snapshot_space_id: space},
+  });
+
+  for (const community of associatedCommunities) {
+    const communityId = community.chain_id;
+    const communityDiscordConfig = await this.models.DiscordBotConfig.findAll({
```

Comment:
Any way we can avoid doing this findAll inside a loop, using e.g. map instead? Ideally we can make this code more concurrent (since we're going to have to wait anyway on the post request).

Reply:
for tim

Reply:
@timolegros

Reply:
bad style but will be ok for the moment